posts/pytables-direct-chunking.rst

```diff
@@ -0,0 +1,131 @@
+    ...     wchunk = b2chunk.to_cframe()  # Serialize.
+    ...     earray.write_chunk((cstart, 0), wchunk)
+
+You can see that such low-level writing is more involved than usual.  Though we used default Blosc2 parameters here, the explicit compression step allows you to fine-tune it in ways not available through PyTables like setting internal chunk and block sizes or even using Blosc2 compression plugins like Grok/JPEG2000.  In fact, the filters given on dataset creation are only used as a hint, since each Blosc2 container holding a chunk includes enough metadata to process it independently.  In the example, the default chunk compression parameters don't even match dataset filters (using Zstd instead of LZ4)::
+
+    >>> carray.filters
+    Filters(complevel=2, complib='blosc2:lz4', ...)
+    >>> earray.filters
+    Filters(complevel=2, complib='blosc2:lz4', ...)
+    >>> b2chunk.schunk.cparams['codec']
+    <Codec.ZSTD: 5>
+
+Still, the Blosc2 HDF5 filter plugin included with PyTables is able to read the data just fine::
+
+    >>> bool((carray[:] == earray[:]).all())
+    True
+    >>> h5f.close()
+
+You may find a more elaborate example of using direct chunking `in PyTables' examples <https://github.com/PyTables/PyTables/blob/master/examples/direct-chunking.py>`_.
+
+Benchmarks
+----------
+
+`b2nd optimized slicing <https://www.blosc.org/posts/pytables-b2nd-slicing/>`_ shows us that removing the HDF5 filter pipeline from the I/O path can result in sizable performance increases, if the right chunking and compression parameters are chosen.  To check the impact of using the new direct chunking API, we ran some benchmarks that compare regular and direct read/write speeds.  On an AMD Ryzen 7 7800X3D CPU with 8 cores, 96 MB L3 cache and 8 MB L2 cache, clocked at 4.2 GHz, we got the following results:
+
+.. image:: /images/pytables-direct-chunking/AMD-7800X3D.png
+  :width: 75%
+  :align: center
+
+We can see that direct chunking yields 3.75x write and 4.4x read speedups, reaching write/read speeds of 1.7 GB/s and 5.2 GB/s.  These are quite impressive numbers, though the base equipment is already quite powerful.  Thus we also tried the same benchmark on a consumer-level MacBook Air laptop with an Apple M1 CPU with 4+4 cores and 12 MB L2 cache, clocked at 3.2 GHz, with the following results:
+
+.. image:: /images/pytables-direct-chunking/MacAir-M1.png
+  :width: 75%
+  :align: center
+
+In this case direct chunking yields 4.5x write and 1.9x read speedups, with write/read speeds of 0.8 GB/s and 1.6 GB/s.  The absolute numbers are of course not as impressive, but the performance is still much better than that of the regular mechanism, especially when writing.  Please note that the M1 CPU has a hybrid efficiency+performance core configuration.  We also run the tests on another hybrid CPU of the high-performance range, in this case an Intel Core i9-13900K CPU with 8+16 cores and 32 MB L2 cache, clocked at 5.7 GHz:
```

Comment:
To be honest, I'd not add so many benchmarks of different machines, as this can overwhelm readers a bit ;-). I'd skip the Intel machine, as it has a similar architecture (and behavior) than the AMD.

Reply:
Yup, I agree, though the write results on the Intel are IMO too good not to mention them, so I replaced the whole passage with a short summary sentence in commit fa0dc00. Does it look ok to you like this?

Reply:
Yes, that's fine.